tools/update/README.md

```diff
@@ -12,17 +12,39 @@ The tool automatically detects versions and supports multiple modules per repo.
 npm i -g typescript ts-node
 ```
 
-### Run
+### Authentication
+
+The tool requires a GitHub token to access the GitHub API. It will automatically try to find a token from several sources (in order):
+
+1. **Environment variables**: `GH_TOKEN` or `GITHUB_TOKEN`
+2. **Git config**: `git config --get github.token`  
+3. **GitHub CLI**: `gh auth token` (if `gh` is installed and authenticated)
+4. **macOS Keychain**: github.com password (macOS only)
 
+#### Setup Options
+
+**Option 1: Environment variable**
 ```sh
-GH_TOKEN=<github_token> ts-node main.ts update --version <version>
+export GH_TOKEN=<your_github_token>
```

Comment:
It's not a good idea to put such a sensitive token into your environment, as now all commands have in the shell have access to it. This is why the old command only passed the environment variable to the invoked command.